File: packages/php-sdk/src/Types/Requests/Quote/UpdateProductRequest.php
<?php

declare(strict_types=1);

namespace TurboDocx\Types\Requests\Quote;

/**
 * Request for updating a product
 */
final class UpdateProductRequest
{
    /**
     * @param string[] $images File paths or raw bytes for product images
     * @param string[]|null $imageIdsToKeep
     * @param string[]|null $imageOrder
     */
    public function __construct(
        public readonly ?string $name = null,
        public readonly ?float $listPrice = null,
        public readonly ?string $billingFrequency = null,
        public readonly ?string $sku = null,
        public readonly ?string $description = null,
        public readonly ?string $detailedSpecification = null,
        public readonly ?string $internalNotes = null,
        public readonly ?string $categoryId = null,
        public readonly ?float $cost = null,
        public readonly ?int $minimumOrderQuantity = null,
        public readonly ?string $currency = null,
        public readonly ?bool $showInCatalog = null,
        public readonly array $images = [],
        public readonly ?array $imageIdsToKeep = null,
        public readonly ?array $imageOrder = null,
    ) {}

    /**
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        $data = [];

        if ($this->name !== null) {
            $data['name'] = $this->name;
        }
        if ($this->listPrice !== null) {
            $data['listPrice'] = $this->listPrice;
        }
        if ($this->billingFrequency !== null) {
            $data['billingFrequency'] = $this->billingFrequency;
        }
        if ($this->sku !== null) {
            $data['sku'] = $this->sku;
        }
        if ($this->description !== null) {
            $data['description'] = $this->description;
        }
        if ($this->detailedSpecification !== null) {
            $data['detailedSpecification'] = $this->detailedSpecification;
        }
        if ($this->internalNotes !== null) {
            $data['internalNotes'] = $this->internalNotes;
        }
        if ($this->categoryId !== null) {
            $data['categoryId'] = $this->categoryId;
        }
        if ($this->cost !== null) {
            $data['cost'] = $this->cost;
        }
        if ($this->minimumOrderQuantity !== null) {
            $data['minimumOrderQuantity'] = $this->minimumOrderQuantity;
        }
        if ($this->currency !== null) {
            $data['currency'] = $this->currency;
        }
        if ($this->showInCatalog !== null) {
            $data['showInCatalog'] = $this->showInCatalog;
        }
        if (count($this->images) > 0) {
            $data['images'] = $this->images;
        }
        if ($this->imageIdsToKeep !== null) {
            $data['imageIdsToKeep'] = $this->imageIdsToKeep;
        }
        if ($this->imageOrder !== null) {
            $data['imageOrder'] = $this->imageOrder;
        }

        return $data;
    }
}

File: packages/java-sdk/src/main/java/com/turbodocx/models/quote/UpdateProductRequest.java
package com.turbodocx.models.quote;

import java.util.List;

/**
 * Request to update a product.
 * Note: images are handled via multipart form data at the TurboQuote layer.
 */
public class UpdateProductRequest {
    private String name;
    private Double listPrice;
    private String billingFrequency;
    private String sku;
    private String description;
    private String detailedSpecification;
    private String internalNotes;
    private String categoryId;
    private Double cost;
    private Integer minimumOrderQuantity;
    private String currency;
    private Boolean showInCatalog;
    private List<String> imageIdsToKeep;
    private List<String> imageOrder;

    // images are handled separately (byte arrays for multipart upload)
    private transient byte[][] images;

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public Double getListPrice() { return listPrice; }
    public void setListPrice(Double listPrice) { this.listPrice = listPrice; }
    public String getBillingFrequency() { return billingFrequency; }
    public void setBillingFrequency(String billingFrequency) { this.billingFrequency = billingFrequency; }
    public String getSku() { return sku; }
    public void setSku(String sku) { this.sku = sku; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getDetailedSpecification() { return detailedSpecification; }
    public void setDetailedSpecification(String detailedSpecification) { this.detailedSpecification = detailedSpecification; }
    public String getInternalNotes() { return internalNotes; }
    public void setInternalNotes(String internalNotes) { this.internalNotes = internalNotes; }
    public String getCategoryId() { return categoryId; }
    public void setCategoryId(String categoryId) { this.categoryId = categoryId; }
    public Double getCost() { return cost; }
    public void setCost(Double cost) { this.cost = cost; }
    public Integer getMinimumOrderQuantity() { return minimumOrderQuantity; }
    public void setMinimumOrderQuantity(Integer minimumOrderQuantity) { this.minimumOrderQuantity = minimumOrderQuantity; }
    public String getCurrency() { return currency; }
    public void setCurrency(String currency) { this.currency = currency; }
    public Boolean getShowInCatalog() { return showInCatalog; }
    public void setShowInCatalog(Boolean showInCatalog) { this.showInCatalog = showInCatalog; }
    public List<String> getImageIdsToKeep() { return imageIdsToKeep; }
    public void setImageIdsToKeep(List<String> imageIdsToKeep) { this.imageIdsToKeep = imageIdsToKeep; }
    public List<String> getImageOrder() { return imageOrder; }
    public void setImageOrder(List<String> imageOrder) { this.imageOrder = imageOrder; }
    public byte[][] getImages() { return images; }
    public void setImages(byte[][] images) { this.images = images; }
}
